src/DSP.jl

```diff
@@ -39,24 +44,63 @@ for (T, suff) in ((Float64, "D"), (Float32, ""))
 end
 
 
-function plan_dct(n,k::Integer)
-    @assert isinteger(Base.log2(n))
-    @assert 2≤k≤4
-    ccall(("vDSP_DCT_CreateSetup",libacc),Ptr{Void},(Ptr{Void},Cint,Cint),C_NULL,n,k)
+## == Discrete Cosine Transform (DCT) == ##
+"""
+Initializes a new DCT setup object. 'dct_type' must be 2, 3, 4 corresponding to Type II, III and IV.
+DCT 'length' must be equal to f*(2^n) where f = 1,3,5,15 and n >= 4. If you have a previous DCT setup
+object, that can be passed in as 'previous'. The returned DCT setup will share the underlying data
+storage of the previous setup object.
+
+Returns: DFTSetup
+"""
+function plan_dct(length::Integer,  dct_type::Integer, previous=C_NULL)
+    if dct_type < 2 &&  dct_type > 4
+        error("DCT type ", dct_type, " is not supported. Only DCT types 2, 3 and 4 are supported")
+    elseif !(isinteger(Base.log2(length)))
```

Comment:
The check wasn't meant to be exhaustive, just a general check for some of the base conditions, but your method is better. I'll change it.